state/in-memory/in_memory.go

```diff
@@ -377,15 +357,16 @@ func (store *inMemoryStore) Multi(ctx context.Context, request *state.Transactio
 
 	// step3: do really set
 	// these operations won't fail
+	var err error
 	for _, o := range request.Operations {
 		switch req := o.(type) {
 		case *innerSetRequest:
-			store.doSet(ctx, req.req.Key, req.data, req.ttl)
+			err = errors.Join(err, store.doSet(ctx, req.req.Key, req.req.Value, req.ttl))
```

Comment:
Why the errors.Join here?

Reply:
We are inside a range, and want to catch all `doSet` errors.

Reply:
However this is not really how `errors.Join` should be used, as the Go team told me too: https://github.com/golang/go/issues/60209
Should collect the errors in a slice and then call `errors.Join` once